### PR TITLE
Adds 4byte registry fallback to getMethodData()

### LINKED ISSUE #5057: Look up function signatures in 4byte registry
Right now we look up funcation signatures using Parity on-chain function registry. If found, we fill the function name on our Confirm screen.

The [4byte registry](https://www.4byte.directory/) is much larger, so we could search it as a fallback option if the function isn't found in Parity's registry.

Major drawback here is privacy (we'd be querying a centralized API with much of a user's transaction data), so we should also consider a feature toggle in settings to let users disable this request.

### PR DESCRIPTION
**Pull request by danjm**
fixes #5057 

This PR adds a call to the 4byte registry as a fallback if our `eth-method-registry` call comes up empty. The work is done in getMethodData. A couple other minor issues that I came across while completing this PR are addressed. They are explained by the commit history.

- [x] This PR increases the time it takes to load the confirm screen. It is possible to avoid that by only blocking the name of the method and associated arg data, and not the whole screen, on the 4byte registry request. We could even go as far as rendering a screen with minimal details even before the eth-method-registry request is resolved. This would help with the already too long loading screen. I may move this into another PR.

### CODE REVIEW

**Comment by Chornuy91** [comment]
chornuy090591@bigmir.net

**Review by whymarrh — CHANGES_REQUESTED** [review]
LGTM, one small thing though we could maybe write `getMethodFrom4Byte` using async await

File: ui/app/helpers/utils/transactions.util.js [hunk 1]
```diff
@@ -43,7 +58,19 @@ const registry = new MethodRegistry({ provider: global.ethereumProvider })
     const fourBytePrefix = prefixedData.slice(0, 10)
 
     try {
-      const sig = await registry.lookup(fourBytePrefix)
+      let fourByteSig
+      try {
+        fourByteSig = getMethodFrom4Byte(fourBytePrefix)
+      } catch (e) {
+        fourByteSig = Promise.resolve(null)
+        console.log(e)
```
Comment (whymarrh) [inline ui/app/helpers/utils/transactions.util.js]: Can we use the `log` module here?

File: ui/app/helpers/utils/transactions.util.js [hunk 2]
```diff
@@ -43,7 +58,19 @@ const registry = new MethodRegistry({ provider: global.ethereumProvider })
     const fourBytePrefix = prefixedData.slice(0, 10)
 
     try {
-      const sig = await registry.lookup(fourBytePrefix)
+      let fourByteSig
+      try {
+        fourByteSig = getMethodFrom4Byte(fourBytePrefix)
```
Comment (whymarrh) [inline ui/app/helpers/utils/transactions.util.js]: I don't quite think that the `try` block will catch any async errors here, we might be able to chain a `catch` onto this here, e.g.,

```js
fourByteSig = getMethodFrom4Byte(fourBytePrefix).catch((e) => {
    log.error(e);
    return null;
})
```

File: ui/app/helpers/utils/transactions.util.js (hunk 1 above)
Comment (whymarrh) [reply ui/app/helpers/utils/transactions.util.js]: Or the comment above addresses this as well

**Review by whymarrh — CHANGES_REQUESTED** [review]

**Comment by danfinlay** [comment]
For the sake of neat abstraction, what would you think of adding this to the `[eth-method-registry](https://github.com/danfinlay/eth-method-registry)` module itself?

**Comment by danfinlay** [comment]
Maybe it wouldn't make sense, since this uses `fetch` and that module currently doesn't assume any particular transport...

File: ui/app/helpers/utils/transactions.util.js (hunk 1 above)
Comment (danjm) [reply ui/app/helpers/utils/transactions.util.js]: @whymarrh Could we rely on the existing try-catch block. So the whole thing would become:

```
    try {
      const fourByteSig = getMethodFrom4Byte(fourBytePrefix)

      let sig = await registry.lookup(fourBytePrefix)

      if (!sig) {
        sig = await fourByteSig
      }

      if (!sig) {
        return {}
      }

      const parsedResult = registry.parse(sig)

      return {
        name: parsedResult.name,
        params: parsedResult.args,
      }
    } catch (error) {
      log.error(error)
      const tokenData = getTokenData(data)
      const { name } = tokenData || {}
      return { name }
    }
```

File: ui/app/helpers/utils/transactions.util.js (hunk 2 above)
Comment (danjm) [reply ui/app/helpers/utils/transactions.util.js]: @whymarrh Could we rely on the existing try-catch block. So the whole thing would become:

```
    try {
      const fourByteSig = getMethodFrom4Byte(fourBytePrefix)

      let sig = await registry.lookup(fourBytePrefix)

      if (!sig) {
        sig = await fourByteSig
      }

      if (!sig) {
        return {}
      }

      const parsedResult = registry.parse(sig)

      return {
        name: parsedResult.name,
        params: parsedResult.args,
      }
    } catch (error) {
      log.error(error)
      const tokenData = getTokenData(data)
      const { name } = tokenData || {}
      return { name }
    }
```

Comment (whymarrh) [reply ui/app/helpers/utils/transactions.util.js]: I think we still need to handle the case where the promise rejects, a case that isn't handled by the existing `try` block. In the event that `getMethodFrom4Byte(fourBytePrefix)` rejects and `registry.lookup(fourBytePrefix)` resolves to something truthy we don't have a handler for the rejected promise which I believe will result in [an unhandled promise rejection](https://stackoverflow.com/a/40500776/1267663).

**Review by whymarrh — APPROVED** [review]
LGTM

**Pull request merged**